pkg/xray/reports.go

```diff
@@ -0,0 +1,802 @@
+	Type                string           `json:"type,omitempty"` // Violations report filter
+	WatchNames          []string         `json:"watch_names,omitempty"`
+	WatchPatterns       []string         `json:"watch_patterns,omitempty"`
+	PolicyNames         []string         `json:"policy_names,omitempty"`
+	Updated             *StartAndEndDate `json:"updated"`
+	SecurityFilters     *SecurityFilter  `json:"security_filters"`
+	LicenseFilters      *LicenseFilter   `json:"license_filters"`
+	Risks               []string         `json:"risks,omitempty"`     // Operational risks filter
+	ScanDate            *StartAndEndDate `json:"scan_date,omitempty"` // Common attributes
+	Component           string           `json:"component,omitempty"`
+	Artifact            string           `json:"artifact,omitempty"`
+	Severities          []string         `json:"severities,omitempty"`
+}
+
+type CvssScore struct {
+	MinScore float64 `json:"min_score,omitempty"`
+	MaxScore float64 `json:"max_score,omitempty"`
+}
+
+type StartAndEndDate struct {
+	Start string `json:"start,omitempty"`
+	End   string `json:"end,omitempty"`
+}
+
+type SecurityFilter struct {
+	Cve             string     `json:"cve,omitempty"`
+	IssueId         string     `json:"issue_id,omitempty"`
+	CvssScore       *CvssScore `json:"cvss_score,omitempty"`
+	SummaryContains string     `json:"summary_contains"`
+	HasRemediation  bool       `json:"has_remediation,omitempty"`
+}
+
+type LicenseFilter struct {
+	Unknown         bool     `json:"unknown"`
+	Unrecognized    bool     `json:"unrecognized"`
+	LicenseNames    []string `json:"license_names,omitempty"`
+	LicensePatterns []string `json:"license_patterns"`
+}
+
+func unpackReport(d *schema.ResourceData, reportType string) *Report {
+	report := Report{}
+
+	if v, ok := d.GetOk("project_key"); ok {
+		report.ProjectKey = v.(string)
+	}
+	report.Name = d.Get("name").(string)
+
+	report.Resources = unpackResources(d.Get("resources").(*schema.Set))
+
+	if reportType == "vulnerabilities" {
+		report.Filters = unpackVulnerabilitiesFilters(d.Get("filters").(*schema.Set))
+	}
+
+	if reportType == "licenses" {
+		report.Filters = unpackLicensesFilters(d.Get("filters").(*schema.Set))
+	}
+
+	if reportType == "violations" {
+		report.Filters = unpackViolationsFilters(d.Get("filters").(*schema.Set))
+	}
+
+	if reportType == "operationalRisks" {
+		report.Filters = unpackOperationalRisksFilters(d.Get("filters").(*schema.Set))
+	}
+
+	return &report
+}
+
+func unpackReportProjectKey(d *schema.ResourceData) *Report {
+	report := Report{}
+
+	if v, ok := d.GetOk("project_key"); ok {
+		report.ProjectKey = v.(string)
+	}
+
+	return &report
+}
+
+func unpackResources(configured *schema.Set) *Resources {
+	var resources Resources
+	m := configured.List()[0].(map[string]interface{})
+
+	if m["repository"] != nil {
+		resources.Repositories = unpackRepository(m["repository"].(*schema.Set))
+	}
+	if m["builds"] != nil {
```

Comment:
Please keep code styling/formatting consistent, i.e. add extra line between `}` and `if ...`, like the subsequent lines in this func.